### ISSUE TITLE
Two RTCRtpSender.replaceTrack tests seem incorrect.

### ISSUE THREAD

**Issue by jan-ivar** [web-platform-tests/wpt#10622]
Two of the tests added in https://github.com/w3c/web-platform-tests/pull/8814 seem incorrect.

 * replaceTrack() rejects when invoked after removeTrack().
 * replaceTrack() rejects after a subsequent removeTrack().

In Firefox these fail with "Expected replaceTrack() to be rejected with InvalidModificationError", but I find no support in the spec for this. A code comment says:
```js
      // replaceTrack() should fail because the sender should be inactive after
      // removeTrack().
```
But the [spec](http://w3c.github.io/webrtc-pc/#dom-rtcrtpsender-replacetrack) says: *"ignore transceiver's [[Direction]] slot in this determination. If negotiation is needed, then reject p with a newly created InvalidModificationError "*

Therefore, it should not matter whether the sender is inactive.

AFAIK replaceTrack is meant to be orthogonal to negotiation, as long as tracks can be compatibly swapped within the same negotiated envelope, if any.

cc @henbos

**Comment by henbos** [comment]
You're right, it should only reject if the transceiver is stopped, but removeTrack() doesn't stop the transceiver it only changes its [[Direction]]. I either mixed up the two concepts of "removing" or the bit about ignoring the [[Direction]] changed since the tests were written.

**Comment by henbos** [comment]
> Determine if negotiation would be needed in order to replace sender's existing track with withTrack. Ignore whether the sender already needs negotiation, in this determination. Negotiation is not needed if withTrack is null. Ignore which MediaStream the track resides in and the id attribute of the track in this determination. Also, ignore transceiver's [[Direction]] slot in this determination. If negotiation is needed, then reject p with a newly created InvalidModificationError and abort these steps.

What does "ignore if the sneder already needs negotiation"? Does it mean "if it already needs negotiation, then don't reject the promise"?

And "ignore transceiver's [[Direction]] slot in this determination"? What does that mean, does that mean to perform [check if negotiation is needed](https://w3c.github.io/webrtc-pc/#dfn-update-the-negotiation-needed-flag) but skip the substeps of the algorithm that talk about [[Direction]]? That would mean to skip these steps:

> 4.2.1 If transceiver's [[Direction]] slot is "sendrecv" or "sendonly", and the associated m= section in description doesn't contain an "a=msid" line, return true.

> 4.2.3 If description is of type "answer", and the direction of the associated m= section in the description does not match transceiver's [[Direction]] slot intersected with the offered direction (as described in [JSEP] (section 5.3.1.)), return true.

**Comment by jan-ivar** [comment]
> What does "ignore if the sender already needs negotiation"? Does it mean "if it already needs negotiation, then don't reject the promise"?

It means determine if the new track can be sent immediately by a negotiated sender, i.e. without violating its negotiated envelope, and if not, reject. I find the green note under [replaceTrack](http://w3c.github.io/webrtc-pc/#dom-rtcrtpsender-replacetrack) clarifying about the kinds of things replaceTrack is meant to reject on. These are all rare (unimplemented AFAIK).

When the sender is not yet negotiated there's no violation (see https://github.com/w3c/webrtc-pc/issues/1855 on this).

I'm open to better ways to describe this. Implying this needs the whole negotiation algorithm, except ignore all these things, is perhaps not the clearest way to express this.

Hope that helps.

**Comment by henbos** [comment]
Yes thank you.

**Comment by henbos** [comment]
I assigned this on myself but I don't know when I will have the time to fix it so anyone that is up for it is free to re-assign to themselves.

**Comment by dontcallmedom** [comment]
closed in #13916
